features/netsocket/emac-drivers/TARGET_ARM_SSG/COMPONENT_SMSC9220/smsc9220_emac.cpp

```diff
@@ -169,12 +178,8 @@ bool SMSC9220_EMAC::link_out(emac_mem_buf_t *buf)
                                       (const char*)_memory_manager->get_ptr(buf),
                                       _memory_manager->get_len(buf));
         _memory_manager->free(buf);
-        if (error != SMSC9220_ERROR_NONE) {
-            _TXLockMutex.unlock();
-            return false;
-        }
         _TXLockMutex.unlock();
-        return true;
+        return (error == SMSC9220_ERROR_NONE)
```

Comment:
Missing `;`!

(and unnecessary parentheses, but they don't hurt).

Reply:
Sorry about that, fixed.